src/main/java/com/puppycrawl/tools/checkstyle/JavaAstVisitor.java

```diff
@@ -558,9 +558,8 @@ public DetailAstImpl visitClassOrInterfaceType(
         for (ParserRuleContext extendedContext : ctx.extended) {
             final DetailAstImpl dot = create(extendedContext.start);
             DetailAstPair.makeAstRoot(currentAST, dot);
-            final List<ParseTree> childList = extendedContext
-                    .children.subList(1, extendedContext.children.size());
```

Comment:
The mutation is basically saying the `subList` doesn't need to be done. The only thing the `subList` is doing is dropping the first item.

The first item in one example is a `period` and a terminal node (from `InputAntlr4AstRegressionWeirdCtor`, code `final Map.Entry<Class<?>, Integer> entry`). The reason this `subList` isn't needed is because the node is a terminal node. When this terminal node is allowed to pass to `visit`, it goes through the specific and default handling of terminal nodes which is to return `null`. When `null` gets to `addAstChild` it basically just drops it, so this terminal node isn't added to the tree there. A few lines above in the visitor's code, you can see the dot is specifically being handled by the `create` method there. So we aren't dropping a tree node, thus there is no harm in removing this code.

Rewriting of the code is the result of the `subList` call not being needed, thus we don't need to create the `childList` variable.